src/ServiceMonitor/IISConfigUtil.cpp

```diff
@@ -144,6 +156,14 @@ HRESULT IISConfigUtil::BuildAppCmdCommand(const vector<pair<wstring, wstring>>&
         wstring strEnvName  = envVecIter->first;
         wstring strEnvValue = envVecIter->second;
 
+        //
+        // Handle values that have single and double quotes
+        //
+        Replace(strEnvName, L"'", L"''");

```

Comment:
This can be done in 1 invocation by using find_first_of instead of find. find_first_of takes an array of chars to search for so you could pass it a array of chars that contains a single quote and a double quote. When you find it, just replace it with a string that contains the same char doubled.

Reply:
Good point. BTW, I found another problem here. We should create a function to handle all the characters which can cause the similar issue while running AppCmd.exe. I will implement a new function for that purpose.